vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SlideManagerTest.java

```diff
@@ -0,0 +1,112 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.vspace.core.data.SequenceRepository;
+import edu.asu.diging.vspace.core.data.SlideRepository;
+import edu.asu.diging.vspace.core.exception.SlideDoesNotExistException;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.Sequence;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+
+public class SlideManagerTest {
+
+	@Mock
+	private SequenceRepository sequenceRepo;
+
+	@Mock
+	private SlideRepository slideRepo;
+
+	@InjectMocks
+	private SlideManager slideManagerToTest = new SlideManager();
+
+	// setting common used variables and Objects
+	String slideId, moduleId, sequenceId, slideIdNotInSequence;
+	List<ISlide> slidesList = Arrays.asList(new Slide());
+	Sequence sequenceObj;
+
+
+	@Before
+	public void init() {
+		MockitoAnnotations.initMocks(this);
+
+		sequenceObj = new Sequence() ;
+		slideId = "SLI000000002";
+		slideIdNotInSequence = "SLI000000219";
+		moduleId = "MOD000000002";
+		sequenceId = "SEQ000000004";
+		slidesList.get(0).setId(slideId);
+	}    
+
+	@Test
+	public void test_slideSequence_slideExistsInSequence() {
+
+		// Positive scenario - Slide present in Sequence
+
+		sequenceObj.setId(sequenceId);
+		sequenceObj.setSlides(slidesList);
+		List<Sequence> sequencesList = new ArrayList<>();
+		sequencesList.add(sequenceObj);
+		Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+		List<Sequence> slideSequencePresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+
+		String actualSlideIdSequence = slideSequencePresent.get(0).getSlides().get(0).getId();
+		String expectedSlideIdSequence = sequencesList.get(0).getSlides().get(0).getId();
+
+		Assert.assertEquals(actualSlideIdSequence, expectedSlideIdSequence);
+		Assert.assertEquals(slideSequencePresent.size(), sequencesList.size()); 
+
+	}
+
+	@Test
+	public void test_slideSequence_slideNotExistsInSequence() {
+
+		List<Sequence> slideSequencePresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+		List<Sequence> actualSequenceSlideListNotPresent = slideManagerToTest.getSlideSequences(slideIdNotInSequence, moduleId);

```

Comment:
you need to make sure that what the manager returns is what you expect. if `getSlideSequences` shouldn't return anything for the provided parameters, that's what you need to test (e.g. returned list is empty or null), not that it is not the same as for ids that should return something. there could be plenty of ways `assertNotSame` would be true but the method is still doing the wrong thing.

Reply:
I am passing slideId which is not in sequence which is returning a null list.
And I am passing slideId which is there in sequence which is returning data.
So, I am checking them.
What other check do I need to make?

Reply:
if `getSlideSequences` should return null for the given id, then you need to assert exactly that.

Reply:
Is Assert.assertNotSame not a proper way to check?

Reply:
It's returning a list of size 0, hence I am checking that.

Reply:
you're not checking if the returned list has size 0 but that the returned list is not the same as a different list. if the returned list has 2 elements and you check if it is not the same as a list that has 1 element the test would pass although the method doesn't do what it's supposed to be doing.

Reply:
Oh okay.